src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -1069,6 +1069,34 @@ int fx_muxer_t::get_runtime_delegate(host_context_t *context, coreclr_delegate_t
 const host_context_t* fx_muxer_t::get_active_host_context()
 {
     std::lock_guard<std::mutex> lock{ g_context_lock };
+    if (g_active_host_context == nullptr)
+        return nullptr;
+
+    if (g_active_host_context->type != host_context_type::empty)
+        return g_active_host_context.get();
+
+    // Try to populate the contract for the 'empty' active context (i.e. created through non-context-based APIs)
+    const hostpolicy_contract_t &hostpolicy_contract = g_active_host_context->hostpolicy_contract;
+    if (hostpolicy_contract.initialize == nullptr)
+    {
+        trace::warning(_X("Getting the contract for the initialized hostpolicy is only supprted for .NET Core 3.0 or a higher version."));
+        return g_active_host_context.get();
+    }
+
+    corehost_context_contract hostpolicy_context_contract;
+    {
+        propagate_error_writer_t propagate_error_writer_to_corehost(hostpolicy_contract.set_error_writer);
+        int rc = hostpolicy_contract.initialize(nullptr, intialization_options_t::get_contract, &hostpolicy_context_contract);
+        if (rc != StatusCode::Success)
+        {
+            trace::warning(_X("Failed to get contract for existing initialized hostpolicy: 0x%x"), rc);
```

Comment:
I would prefer to fail here. I understand we can make it work, but I think it's dangerous to ignore any failure from hostpolicy. I can't think of a scenario where having this would be important, so unless we can come up with one, I would prefer we block this branch for now. We can always reenable it later on.